src/core/public/application/ui/app_not_found_screen.tsx

```diff
@@ -53,6 +57,8 @@ export const AppNotFound = () => (
                 id="core.application.appNotFound.pageDescription"
                 defaultMessage="No application was found at this URL. Try going back or choosing an app from the menu."
               />
+              <br />
+              <a href={baseUrl + '/app/home#/'}>Go to Home Page</a>
```

Comment:
I tried using `FormattedHTMLMessage` component to modify existing message to contain the re-direct link but was unsuccessful

Reply:
There's a `defaultRoute` in advanced settings that's user configurable. We should use that instead of hard-coding `/app/home#/`

![Screen Shot 2022-04-25 at 3 40 34 PM](https://user-images.githubusercontent.com/3443142/165186384-960f1301-391f-4a2d-b8b1-ce1b3c2bd440.png)

Reply:
I'm unable to pinpoint the exact location where I can reference the `defaultRoute` variable from. The image that you shared loads this as part of advancedSettings on page load and all the settings are stored in an array. I looked in other files too and they load defaultRoute on file load. See [here](https://github.com/DarshitChanpura/OpenSearch-Dashboards/blob/c17c88dff4a0d7f1417275cd6af26c45c3b41978/src/core/server/core_app/core_app.ts#L54)
I may be completely off track here.

Would appreciate some help!

Reply:
That is correct. The correct way to access the uiSettings right now is using the `client.get` api.

```ts
await core.uiSettings.client.get<string>('defaultRoute')
```

@kavilla @tmarkley In the example @DarshitChanpura mentions, we also have a BWC condition to handle for the `kibana_overview` page, is that still necessary?

Reply:
I believe we do want to retain that backwards compatibility behavior.

Reply:
Can we add a test to confirm that the link is created as expected? Also, should we use the same `landing page` language that is used in the advanced setting description instead of `home page`?

Reply:
I plus one to landing page? I just confirmed that removing `/app/home` will and relying on baseUrl will hit the logic that will re-route to the defaultRoute, so this should be supported. Just the text should be updated and should support i18n.